src/bloom/bit-set.ts

```diff
@@ -0,0 +1,168 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+import {encode, decode} from "base64-arraybuffer";
+
+const bitsPerWord = 8;
+
+/** A memory-efficient Boolean array. Contains just the minimal operations needed for our Bloom filter implementation.
+ *
+ * @author David Leppik
+ */
+export default class BitSet {
+    readonly size: number;
+
+    // Uint32Array may be slightly faster due to memory alignment, but this avoids endianness when serializing
+    private array: Uint8Array;
+
+    /**
+     * Constructor. All bits are initially set to false.
+     * @param size the number of bits that can be stored. (This is NOT required to be a multiple of 8.)
+     */
+    constructor(size: number) {
+        this.size = size
+        this.array = new Uint8Array(Math.ceil(size / bitsPerWord))
+    }
+
+    /** Returns the value of the bit at the given index
+     * @param index position of the bit, zero-indexed
+     */
+    has(index: number): boolean {
+        const wordIndex = Math.floor(index / bitsPerWord)
+        const mask = 1 << (index % bitsPerWord)
+        return (this.array[wordIndex] & mask) !== 0
+    }
+
+    /** Set the bit to true
+     * @param index position of the bit, zero-indexed
+     */
+    add(index: number) {
+        const wordIndex = Math.floor(index / bitsPerWord)
+        const mask = 1 << (index % bitsPerWord)
+        this.array[wordIndex] = this.array[wordIndex] | mask
+    }
+
+    /** Set the bit to false
+     * @param index position of the bit, zero-indexed
+     */
+    remove(index: number) {
+        const wordIndex = Math.floor(index / bitsPerWord)
+        const mask = 1 << (index % bitsPerWord)
+        this.array[wordIndex] = this.array[wordIndex] ^ mask
+    }
+
+    /** Returns the maximum true bit. */
+    max(): number {
+        for (let i = this.array.length - 1; i >= 0; i--) {
+            let bits = this.array[i];
+            if (bits) {
+                return BitSet.highBit(bits) + (i*bitsPerWord);
+            }
+        }
+        return 0;
+    }
+
+    /** Returns the number of true bits. */
+    bitCount(): number {
+        let result = 0
+        for (let i = 0; i < this.array.length; i++) {
+            result += BitSet.countBits(this.array[i]) // Assumes we never have bits set beyond the end
+        }
+        return result
+    }
+
+    /**
+     * Returns true if the size and contents are identical.
+     * @param other another BitSet
+     */
+    equals(other: BitSet): boolean {
+        if (other.size !== this.size) {
+            return false
+        }
+        for (let i = 0; i < this.array.length; i++) {
+            if (this.array[i] !== other.array[i]) {
+                return false
+            }
+        }
+        return true
+    }
+
+    /** Returns a JSON-encodable object readable by {@link import}. */
```

Comment:
```suggestion
    /** 
     * Returns a JSON-encodable object readable by {@link import}. 
     */
```
Can you fix documentation style for all functions/methods? 
This just style, but will match the rest of the documentation style we want to keep.